core/store/migrate/migrations/0156_bhs_multiple_from_addresses.sql

```diff
@@ -0,0 +1,15 @@
+-- +goose Up
+ALTER TABLE blockhash_store_specs ADD COLUMN from_addresses bytea[] DEFAULT '{}' NOT NULL ;
+
+UPDATE blockhash_store_specs SET from_addresses = from_addresses || from_address
+WHERE from_address IS NOT NULL;
+
+ALTER TABLE blockhash_store_specs DROP COLUMN from_address;
+
+-- +goose Down
+ALTER TABLE blockhash_store_specs ADD COLUMN from_address bytea;
+
+UPDATE blockhash_store_specs SET from_address = from_addresses[1]
+WHERE array_length(from_addresses, 1) > 0;
```

Comment:
Idk about removing the fromAddress field in the same migration as adding the from_addresses field. Is there a graceful way to update the job spec at the same time as a node version upgrade? Otherwise, this will just break the job until we update the job spec to the correct field.

Reply:
Yeah I had the same question as I was working on this change. This migration shouldn't break existing jobs as it will take create the new column and copy any existing address from `fromAddress`. If there is no existing field, then the migration just creates an empty `fromAddresses` field. 

Specifying `fromAddress` in the job spec will not work after this change though. We could update the job specs to have `fromAddresses` after this migration is applied. Otherwise, it will automatically default to first sending key on the node.

Reply:
Yeah I guess that works. Will re-read the PR to look at the tests for this.

Reply:
See https://github.com/smartcontractkit/chainlink/blob/develop/core/store/migrate/migrations/0107_vrf_multiple_from_addresses.sql for the prior art, we did this for v2 jobs.